test/functionalTest/cases/950_GET_v2_entity/GET_v2_entity_complex.test

```diff
@@ -0,0 +1,356 @@
+
+--NAME--
+GET Entity V2 Complex
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. GET /v2/entities
+# 02. POST /v1/updateContext
+# 03. GET /v2/entities
+# 04. GET /v2/entities?count=true
+# 05. POST /v1/updateContext with compound attribute value
+#
+
+echo "01. GET /v2/entities"
+echo "===================="
+orionCurl --url /v2/entities/averyfarwayentitythatdoesnotexist --json
+echo
+echo
+
+
+echo "02. POST /v1/updateContext"
+echo "=========================="
+payload='{
+  "contextElements": [
+    {
+      "id":   "E1",
+      "type": "T1",
+      "attributes" : [
+        {
+          "name" : "A0",
+          "type": "number",
+          "value" : 255
+        },
+        {
+          "name" : "A1",
+          "value" : "E1/T1/A1"
+        },
+        {
+          "name": "A2",
+          "value": "E1/T1/A2",
+          "type": "AT2"
+        },
+        {
+          "name": "A3",
+          "type": "a",
+          "value": "E1/T1/A3/M1",
+          "metadatas": [
+            {
+              "name": "m1",
+              "value": "M1"
+            }
+          ]
+        },
+        {
+          "name": "A4",
+          "type": "a4",
+          "value": "E1/T1/A4/M1-3",
+          "metadatas": [
+            {
+              "name": "m1",
+              "value": "M1"
+            },
+            {
+              "name": "m2",
+              "type": "tm2",
+              "value": "M2"
+            },
+            {
+              "name": "m3",
+              "type": "number",
+              "value": 3
+            },
+            {
+              "name": "type",
+              "value": "TYPE"
+            },
+            {
+              "name": "value",
+              "value": "VALUE"
+            }
+          ]
+        },
+        {
+          "name": "id",
+          "value": "ID"
+        },
+        {
+          "name": "type",
+          "type": "VALUE",
+          "value": "TYPE"
+        }
+      ]
+    }
+  ],
+  "updateAction": "APPEND"
+}'
+orionCurl --url /v1/updateContext --payload "$payload" --json
+echo
+echo
+
+
+echo "03. GET /v2/entities"
+echo "===================="
+orionCurl --url /v2/entities/E1 --json
+echo
+echo
+
+
+echo "04. POST /v1/updateContext with compound attribute value"
+echo "========================================================"
+payload='{
+  "contextElements": [
+    {
+      "id":   "E2",
+      "type": "T1",
+      "attributes" : [
+        {
+          "name" : "A0",
+          "value" : { "i1": 1, "s1": "s" }
+        }
+      ]
+    }
+  ],
+  "updateAction": "APPEND"
+}'
+orionCurl --url /v1/updateContext --payload "$payload" --json
+echo
+echo
+
+
+echo "05. GET /v2/entities"
+echo "===================="
+orionCurl --url /v2/entities/E2 --json
+echo
+echo
+
+
+--REGEXPECT--
+01. GET /v2/entities
+====================
+HTTP/1.1 404 Not Found
+Content-Length: 36
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "error": "No context element found"
```

Comment:
Currently, these error messages are generated by the `httpStatusCodeString(HttpStatusCode code)` function in HttpStatusCode.cpp. Thus, if we want DOMError-style names for v2, it could be easy to provide a similar version to do a new code-to-string mapping. @jmcanterafonseca would you like to do that? ;)

Regarding description, it is ok with me to have that test. I think is a matter of adjusting the render() for the new Entity class. The only problem would be to align this message with the one used by standard operations in the same situation (they would be different).